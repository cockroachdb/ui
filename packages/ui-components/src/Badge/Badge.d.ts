import { FunctionComponent } from "react";
declare type BadgeCase = "none" | "uppercase";
declare type BadgeIntent = "neutral" | "success" | "warning" | "danger";
interface BadgeProps {
    intent?: BadgeIntent;
    transformCase?: BadgeCase;
}
declare const Badge: FunctionComponent<BadgeProps>;
export default Badge;
