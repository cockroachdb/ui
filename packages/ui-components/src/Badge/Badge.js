"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const bind_1 = __importDefault(require("classnames/bind"));
const objectToClassnames_1 = __importDefault(require("../utils/objectToClassnames"));
const Badge_module_scss_1 = __importDefault(require("./Badge.module.scss"));
const cx = bind_1.default.bind(Badge_module_scss_1.default);
const Badge = (_a) => {
    var { intent = "neutral", transformCase = "uppercase", children } = _a, props = __rest(_a, ["intent", "transformCase", "children"]);
    const classnames = cx("badge", objectToClassnames_1.default({ intent, transformCase }));
    if (children !== undefined) {
        return (react_1.default.createElement("div", Object.assign({ className: classnames }, props), children));
    }
    return null;
};
exports.default = Badge;
//# sourceMappingURL=Badge.js.map