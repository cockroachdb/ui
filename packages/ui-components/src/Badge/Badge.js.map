{"version":3,"file":"Badge.js","sourceRoot":"","sources":["Badge.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;AAAA,kDAAiD;AACjD,2DAAyC;AAEzC,qFAA6D;AAE7D,4EAAyC;AAUzC,MAAM,EAAE,GAAG,cAAU,CAAC,IAAI,CAAC,2BAAM,CAAC,CAAC;AAEnC,MAAM,KAAK,GAAkC,CAAC,EAK7C,EAAE,EAAE;QALyC,EAC5C,MAAM,GAAG,SAAS,EAClB,aAAa,GAAG,WAAW,EAC3B,QAAQ,OAET,EADC,2DAAQ;IAER,MAAM,UAAU,GAAG,EAAE,CAAC,OAAO,EAAE,4BAAkB,CAAC,EAAE,MAAM,EAAE,aAAa,EAAE,CAAC,CAAC,CAAC;IAE9E,IAAI,QAAQ,KAAK,SAAS,EAAE;QAC1B,OAAO,CACL,qDAAK,SAAS,EAAE,UAAU,IAAM,KAAK,GAClC,QAAQ,CACL,CACP,CAAC;KACH;IAED,OAAO,IAAI,CAAC;AACd,CAAC,CAAC;AAEF,kBAAe,KAAK,CAAC"}