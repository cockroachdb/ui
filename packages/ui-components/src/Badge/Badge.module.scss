@import "../styles/tokens.scss";

.badge {
  display: inline-block;
  width: fit-content;
  padding: crl-gutters(0.25) crl-gutters(0.75);
  border-radius: 3px;
  vertical-align: crl-gutters(0.5);
  text-align: center;
  white-space: nowrap;
}

.intent--neutral {
  background: $crl-neutral-2;
  color: $crl-base-text--dark;
}

.intent--success {
  background: rgba($crl-green-1, 0.3);
  color: $crl-base-green;
}

.intent--warning {
  background: rgba($crl-orange-1, 0.3);
  color: $crl-base-yellow;
}

.intent--danger {
  background: rgba($crl-base-red, 0.3);
  color: $crl-base-red;
}

.noTransform {
  text-transform: none;
}

.transformCase--none {
  text-transform: none;
}

.transformCase--uppercase {
  text-transform: uppercase;
}
