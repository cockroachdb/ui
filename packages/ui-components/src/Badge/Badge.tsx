import React, { FunctionComponent } from "react";
import classNames from "classnames/bind";

import styles from "./Badge.module.scss";

type BadgeCase = "none" | "uppercase";
type BadgeIntent = "neutral" | "success" | "warning" | "danger";

export interface BadgeProps {
  content?: string;
  intent?: BadgeIntent;
  transformCase?: BadgeCase;
}

const cx = classNames.bind(styles);

const Badge: FunctionComponent<BadgeProps> = ({
  content,
  intent = "neutral",
  transformCase = "uppercase",
  children,
  ...props
}) => {
  const classnames = cx("badge", {
    "intent--success": intent === "success",
    "intent--danger": intent === "danger",
    "intent--warning": intent === "warning",
    "intent--neutral": intent === "neutral",
    "transformCase--uppercase": transformCase === "uppercase",
    "transformCase--none": transformCase === "none",
  });

  if (children !== undefined) {
    return (
      <div className={classnames} {...props}>
        {children}
      </div>
    );
  }

  if (content !== undefined) {
    return (
      <div className={classnames} {...props}>
        {content}
      </div>
    );
  }

  return null;
};

export default Badge;
