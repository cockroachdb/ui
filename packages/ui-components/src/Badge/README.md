[back to components](../README.md)

# Badge

A Badge is a styled text element used to highlight informative elements or act as a label for other visual elements.

```javascript
import React from "react";
import { Badge } from "@cockroachlabs/ui-components";

export const SomeExampleReact = () => (
  <Badge content="Sample 1" />
  <Badge>Sample 2</Badge>
  <Badge content="Successful sample" intent="success" />
  <Badge content="Dangerous sample" intent="danger" />
  <Badget transformCase="none">Meaningful Case Sample v10.3.7-Alpha.0876b</Badge>
);
```

## Properties

These are the properties that can be provided to the `Badge` component.

### `children?: ReactNode`

The content displayed inside the Badge.

This prop is **optional**.

### `content?: String`

A shorthand property used to display content inside the sample

This prop is **optional**.

### `intent?: Enum("neutral", "success", "warning", "danger")`

**default value: `neutral`**

Intent will alter the visual style of the Badge by changing the `background-color` and `color` properties of the rendered element.

This prop is **optional**.

### `transformCase?: Enum("uppercase", "none")`

**default value: `uppercase`**

This prop alters the letter case of the content of the Badge. The default value is uppercase, but can be set to `none` if the case of content is meaningful and should not be transformed.

This prop is **optional**.
