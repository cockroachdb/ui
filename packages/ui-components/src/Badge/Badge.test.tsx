import React from "react";
import { shallow } from "enzyme";

import Badge, { BadgeProps } from "./Badge";

const defaultProps = {
  content: "foo",
};

const renderSubject = (overrideProps?: BadgeProps) => {
  const props = Object.assign(defaultProps, overrideProps);
  return shallow(<Badge {...props} />);
};

describe("Badge content", () => {
  test("can render using content prop", () => {
    const wrapper = renderSubject({ content: "test" });
    expect(wrapper.prop("children")).toBe("test");
  });

  test("can render using children", () => {
    const wrapper = shallow(<Badge>child content</Badge>);
    expect(wrapper.prop("children")).toBe("child content");
  });

  test("rendered by children should override content from the content prop", () => {
    const wrapper = shallow(
      <Badge content="prop content">child content</Badge>,
    );
    expect(wrapper.prop("children")).toBe("child content");
  });

  test("if empty shouldn't render", () => {
    const wrapper = shallow(<Badge />);
    expect(wrapper.isEmptyRender()).toBe(true);
  });
});

describe("Badge Intent prop", () => {
  test("will be neutral by default", () => {
    const wrapper = renderSubject({ content: "intent test" });
    expect(wrapper.prop("className")).toContain("intent--neutral");
  });

  test("will change the display of a badge", () => {
    const wrapper = renderSubject({ content: "intent test" });

    // success
    wrapper.setProps({ intent: "success" });
    expect(wrapper.prop("className")).toContain("intent--success");

    // warning
    wrapper.setProps({ intent: "warning" });
    expect(wrapper.prop("className")).toContain("intent--warning");

    // danger
    wrapper.setProps({ intent: "danger" });
    expect(wrapper.prop("className")).toContain("intent--danger");

    // neutral
    wrapper.setProps({ intent: "neutral" });
    expect(wrapper.prop("className")).toContain("intent--neutral");
  });
});

test("A badge can render the text case provided by content via a prop", () => {
  const wrapper = renderSubject({ transformCase: "none" });
  expect(wrapper.prop("className")).toContain("transformCase--none");
});
