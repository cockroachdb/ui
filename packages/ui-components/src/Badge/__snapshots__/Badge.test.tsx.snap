// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge content can render using children 1`] = `
<div
  className="badge intent--neutral transformCase--uppercase"
>
  child content
</div>
`;

exports[`Badge content can render using content prop 1`] = `
<div
  className="badge intent--neutral transformCase--uppercase"
>
  test
</div>
`;

exports[`Badge content if empty shouldn't render 1`] = `""`;

exports[`Badge content rendered by children should override content from the content prop 1`] = `
<div
  className="badge intent--neutral transformCase--uppercase"
>
  child content
</div>
`;
